redbot/cogs/mutes/voicemutes.py

```diff
@@ -58,98 +66,161 @@ async def _voice_perm_check(
+                else:
+                    issue_list.append((user, success["reason"]))
+
+            if success_list:
+                msg = _("{users} has been muted in this channel{time}.")
+                if len(success_list) > 1:
+                    msg = _("{users} have been muted in this channel{time}.")
+                await ctx.send(
+                    msg.format(users=humanize_list([f"{u}" for u in success_list]), time=time)
                 )
-        else:
-            await ctx.send(issue)
+            if issue_list:
+                msg = _("The following users could not be muted\n")
+                for user, issue in issue_list:
+                    msg += f"{user}: {issue}\n"
+                await ctx.send_interactive(pagify(msg))
 
     @commands.command(name="voiceunmute")
     @commands.guild_only()
     async def unmute_voice(
-        self, ctx: commands.Context, user: discord.Member, *, reason: str = None
+        self,
+        ctx: commands.Context,
+        users: commands.Greedy[discord.Member],
+        *,
+        reason: Optional[str] = None,
     ):
         """Unmute a user in their current voice channel."""
-        user_voice_state = user.voice
-        if (
-            await self._voice_perm_check(
-                ctx, user_voice_state, mute_members=True, manage_channels=True
-            )
-            is False
-        ):
-            return
-        guild = ctx.guild
-        author = ctx.author
-        channel = user_voice_state.channel
-        audit_reason = get_audit_reason(author, reason)
-
-        success, message = await self.unmute_user(guild, channel, author, user, audit_reason)
-
-        if success:
-            await modlog.create_case(
-                self.bot,
-                guild,
-                ctx.message.created_at,
-                "vunmute",
-                user,
-                author,
-                reason,
-                until=None,
-                channel=channel,
-            )
-            await ctx.send(
-                _("Unmuted {user} in channel {channel.name}").format(user=user, channel=channel)
-            )
-            try:
-                if channel.permissions_for(ctx.me).move_members:
-                    await user.move_to(channel)
+        if not users:
+            return await ctx.send_help()
+        if ctx.me in users:
+            return await ctx.send(_("You cannot unmute me."))
+        if ctx.author in users:
+            return await ctx.send(_("You cannot unmute yourself."))
+        async with ctx.typing():
+            issue_list = []
+            success_list = []
+            for user in users:
+                user_voice_state = user.voice
+                if (
+                    await self._voice_perm_check(
+                        ctx, user_voice_state, mute_members=True, manage_channels=True
+                    )
+                    is False
+                ):
+                    continue
```

Comment:
`_voice_perm_check()` sends an error message immediately which doesn't really play that well with how the rest of the command behave (as in, it returns errors at the end for the rest)